cli/command/formatter/stats.go

```diff
@@ -37,11 +36,72 @@ type ContainerStatsAttrs struct {
 	PidsCurrent      uint64 // Not used on Windows
 }
 
-// ContainerStats represents the containers statistics data.
+// ContainerStatsRecord represents the statistics data collected from a container.
+type ContainerStatsRecord struct {
+	name     string
+	windows  bool
+	notValid bool
+	Statistics
+}
+
+// ContainerStats represents the container statistics for collecting.
 type ContainerStats struct {
-	Mu sync.RWMutex
-	ContainerStatsAttrs
-	Err error
+	Windows bool
+	Name    string
+	mutex   sync.Mutex
+	Statistics
+	err error
+}
+
+// GetError returns the container statistics error.
+// This is used to determine whether the statistics are valid or not
+func (cs *ContainerStats) GetError() error {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	return cs.err
+}
+
+// SetErrorAndReset zeroes all the container statistics and store the error.
+// It is used when receiving time out error during statistics collecting to reduce lock overhead
+func (cs *ContainerStats) SetErrorAndReset(err error) {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	cs.CPUPercentage = 0
+	cs.Memory = 0
+	cs.MemoryPercentage = 0
+	cs.MemoryLimit = 0
+	cs.NetworkRx = 0
+	cs.NetworkTx = 0
+	cs.BlockRead = 0
+	cs.BlockWrite = 0
+	cs.PidsCurrent = 0
+	cs.err = err
+}
+
+// SetError sets container statistics error
+func (cs *ContainerStats) SetError(err error) {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	cs.err = err
+}
+
+// SetStatistics set the container statistics
+func (cs *ContainerStats) SetStatistics(s *Statistics) {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	cs.Statistics = *s
+}
+
+// GetStatistics returns container statistics with other meta data such as the container name
+func (cs *ContainerStats) GetStatistics() ContainerStatsRecord {
```

Comment:
Yes, that's right. Nevertheless, you eventually get a `ContainerStatsRecord` entity.
You want me to rename this anyway?